pennylane/qchem/factorization.py

```diff
@@ -14,39 +14,107 @@
 """
 This module contains the functions needed for two-electron tensor factorization.
 """
-
 from functools import partial
 
 import numpy as np
 import scipy as sp
 
 import pennylane as qml
 
-# pylint: disable=too-many-arguments
-
-
-def factorize(two_electron, tol_factor=1.0e-5, tol_eigval=1.0e-5, cholesky=False):
+has_jax_optax = True
+try:  # pragma: no cover
+    # pylint: disable=unused-import
+    import optax
+    from jax import jit
+    from jax import numpy as jnp
+    from jax import scipy as jsp
+    from jax import value_and_grad
+except (ModuleNotFoundError, ImportError) as e:  # pragma: no cover
+    has_jax_optax = False
+
+# pylint: disable=too-many-arguments, too-many-positional-arguments
+
+
+def factorize(
+    two_electron,
+    tol_factor=1.0e-5,
+    tol_eigval=1.0e-5,
+    cholesky=False,
+    compressed=False,
+    regularization=None,
+    **compression_kwargs,
```

Comment:
Small user feedback:

I made a mistake here which "fails" silently.
I used a `regularized=True` keyword argument instead of `regularization='L1'`.

```
symbols  = ['H', 'H']
geometry = np.array([[0.0, 0.0, 0.0],
                     [1.0, 0.0, 0.0]], requires_grad=False)
mol = qml.qchem.Molecule(symbols, geometry)
core, one, two = qml.qchem.electron_integrals(mol)()
two = np.swapaxes(two, 1, 3) # convert to chemist notation
factors, cores, leaves = factorize(two, 1e-5, 1e-5, compressed=True, regularized = True)
print(factors)
```

Do we need any kind of support or checks here or do we just leave it to the user to debug/make sure they read the documentation correctly?

Reply:
Nice catch, added a test for this!